dandi/delete.py

```diff
@@ -0,0 +1,220 @@
+        self.set_dandiset(api_url, dandiset_id)
+        if asset_path.endswith("/"):
+            self.register_asset_folder(api_url, dandiset_id, "draft", asset_path)
+        else:
+            self.register_asset(api_url, dandiset_id, "draft", asset_path)
+
+    def confirm(self) -> bool:
+        if self.deleting_dandiset:
+            msg = f"Delete Dandiset {self.dandiset_id}?"
+        else:
+            msg = (
+                f"Delete {len(self.remote_assets)} assets on server from"
+                f" Dandiset {self.dandiset_id}?"
+            )
+        return click.confirm(msg)
+
+    def delete_dandiset(self) -> None:
+        if self.deleting_dandiset:
+            self.client.delete_dandiset(self.dandiset_id)
+        else:
+            raise RuntimeError(
+                "delete_dandiset() called when Dandiset not registered for deletion"
+            )
+
+    def _process_asset(self, asset: RemoteAsset) -> Iterator[dict]:
+        yield {"status": "Deleting"}
+        try:
+            self.client.delete_asset(
+                asset.dandiset_id, asset.version_id, asset.asset_id
+            )
+        except Exception as e:
+            yield {"status": "Error", "message": f"{type(e).__name__}: {e}"}
+        else:
+            yield {"status": "Deleted"}
+
+    def process_assets_pyout(self) -> Iterator[dict]:
+        with self.client.session():
+            for asset in sorted(self.remote_assets, key=attrgetter("path")):
+                yield {
+                    "path": asset.path,
+                    ("status", "message"): self._process_asset(asset),
+                }
+
+    def process_assets_debug(self) -> Iterator[Iterator[dict]]:
+        with self.client.session():
+            for asset in sorted(self.remote_assets, key=attrgetter("path")):
+                yield ({"path": asset.path, **d} for d in self._process_asset(asset))
+
+
+def delete(
+    paths: Iterable[str],
+    dandi_instance: str = "dandi",
+    devel_debug: bool = False,
+    jobs: Optional[int] = None,
+    force: bool = False,
+) -> None:
+    deleter = Deleter()
+    for p in paths:
+        if is_url(p):
+            deleter.register_url(p)
+        else:
+            deleter.register_local_path_equivalent(dandi_instance, p)
+    if deleter and (force or deleter.confirm()):
+        if deleter.deleting_dandiset:
+            deleter.delete_dandiset()
+        elif devel_debug:
+            for gen in deleter.process_assets_debug():
+                for r in gen:
+                    print(r, flush=True)
+        else:
+            from .support import pyout as pyouts
+
+            pyout_style = pyouts.get_style(hide_if_missing=False)
+            rec_fields = ("path", "status", "message")
+            out = pyouts.LogSafeTabular(
+                style=pyout_style, columns=rec_fields, max_workers=jobs
+            )
+            with out:
+                for gen in deleter.process_assets_pyout():
+                    out(gen)
```

Comment:
please add at least a basic smoke test run with pyout as a backend to ensure that  no gotchas with pyout rendering etc,so we do not miss some breakage